### PR TITLE
Inputs & Outputs

### LINKED ISSUE #54: Implement inputs & outputs group
This requires nested lists concept.

Cf. the [prototype](https://fluffy-fiesta-212e1c7c.pages.github.io/prototypes/visual-ide/v14/).

### Inputs

* String or expression
image

* <del>To be discussed: do we want to rebuild the custom auto complete component or wait for a proper code editor component?</del> ==> no, cf. https://github.com/bpmn-io/bpmn-properties-panel/issues/54#issuecomment-913479908
image

* Script
image

* List
image

* Map
image

<hr></hr>

### Outputs

* String or expression
image

* Script
image

* List
image

* Map
image

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #54 

One follow up PR for the camunda-bpmn-js modeling behaviors: https://github.com/camunda/camunda-bpmn-js/issues/39

-----

String or expression

image image

Script

image image

Map

image image

List

image image

### CODE REVIEW

File: src/provider/camunda-platform/properties/InputProps.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  const {
+    children
+  } = props;
+
+  const {
+    selectedElement: element
+  } = useContext(BpmnPropertiesPanelContext);
+
+  const bpmnFactory = useService('bpmnFactory');
+
```
Comment (MaxTru) [inline src/provider/camunda-platform/properties/InputProps.js:109]:
```suggestion
```

**Review by MaxTru — COMMENTED** [review]
LGTM - can be merged

File: src/provider/camunda-platform/properties/MapProps.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  const entries = map.get('entries');
+
+  function renderEntry(entry, index, open) {
+    const entryId = `${idPrefix}-mapEntry-${index}`;
+
+    return (
+      <Collapsible
+        id={ entryId }
+        entries={ MapEntry({ idPrefix: entryId, element, entry }) }
+        label={ entry.get('key') || '<empty>' }
```
Comment (MaxTru) [inline src/provider/camunda-platform/properties/MapProps.js:38]:
```suggestion
        label={ entry.get('key') || translate('<empty>') }
```

**Comment by pinussilvestrus** [comment]
Thanks for the review @MaxTru, I updated the two minor things.

**Review by andreasgeier — APPROVED** [review]
✅ &nbsp; LGTM

I reviewed based on the screenshots. The special list for the list properties does not look as expected, but we'll fix this in the design finishing together with other things.

**Review by MaxTru — APPROVED** [review]

**Pull request merged**